common/types/map.go

```diff
@@ -94,6 +94,24 @@ func NewProtoMap(adapter Adapter, value *pb.Map) traits.Mapper {
 	}
 }
 
+// NewMutableMap constructs a mutable map from an adapter and a set of map values.
+func NewMutableMap(adapter Adapter, mutableValues map[ref.Val]ref.Val) traits.MutableMapper {
```

Comment:
What happens if a mutable map is part of `mutableValues`? is that possible?

Reply:
That shouldn't be possible since the nested comprehensions will ensure the outputs are immutable. Unless, of course, someone breaks a lot of assumptions with their own extension.